extensions/skyportal/skyportal/handlers/api/archive.py

```diff
@@ -285,6 +291,251 @@ def get(self):
+                    light_curve_ids = [
+                        item["_id"]
+                        for item in response.get("data")[catalog]["query_coords"]
+                    ]
+                    if len(light_curve_ids) == 0:
+                        return self.success(data=[])
+
+                    # return features for the first ID
+                    filter = {"_id": {"$in": [light_curve_ids[0]]}}
+                    query = {
+                        "query_type": "find",
+                        "query": {
+                            "catalog": catalog,
+                            "filter": filter,
+                            "projection": {
+                                "ad": 1,
+                                "chi2red": 1,
+                                "i60r": 1,
+                                "i70r": 1,
+                                "i80r": 1,
+                                "i90r": 1,
+                                "inv_vonneumannratio": 1,
+                                "iqr": 1,
+                                "median": 1,
+                                "median_abs_dev": 1,
+                                "n": 1,
+                                "norm_excess_var": 1,
+                                "norm_peak_to_peak_amp": 1,
+                                "roms": 1,
+                                "skew": 1,
+                                "smallkurt": 1,
+                                "stetson_j": 1,
+                                "stetson_k": 1,
+                                "sw": 1,
+                                "welch_i": 1,
+                                "wmean": 1,
+                                "wstd": 1,
+                                "_id": 1,
+                                "field": 1,
+                                "ccd": 1,
+                                "quad": 1,
+                            },
+                        },
+                    }
+
+                    response = gloria.query(query=query)
+                    if response.get("status", "error") == "success":
+                        features = response.get("data")[0]
+
+                        annotations = []
+                        annotation_data = {}
+                        for key in features.keys():
+                            value = features[key]
+                            if not pd.isnull(value):
+                                if key in ["_id", "field", "ccd", "quad", "n"]:
+                                    value = int(value)
+                                annotation_data[key] = value
+
+                        if annotation_data:
+                            annotation = Annotation(
+                                data=annotation_data,
+                                obj_id=obj_id,
+                                origin=catalog,
+                                author=author,
+                                groups=groups,
```

Comment:
Are you sure the right thing is to post different annotations for each feature? Wouldn't it be better to post one annotation with multiple keys?

Reply:
If the annotations all have the same origin I'm not sure you'd be able to post all of them (the origin+obj_id should be unique).

Reply:
Thanks for the review @guynir42 - the code currently posts a single annotation containing all the different feature keys (see below). If I click the button a second time, it gives the expected 'Annotation already posted' message. 

<img width="596" alt="Screen Shot 2023-02-13 at 10 59 14 AM" src="https://user-images.githubusercontent.com/42810347/218522656-07ce63fd-ad38-4a0d-b2bc-69b6e5b9b500.png">

Reply:
OK I must have misread the loop and conditional indents! Looks good to me.